libsolidity/ast/ASTAnnotations.h

```diff
@@ -208,6 +208,8 @@ struct ExpressionAnnotation: ASTAnnotation
 	bool isLValue = false;
 	/// Whether the expression is used in a context where the LValue is actually required.
 	bool lValueRequested = false;
+	/// Whether the expression is an ordinary assignment (e.g. false for +=, ++, --, .., but true for +=, =, -=, ..)
+	bool ordinaryLAssignment = true;
```

Comment:
```suggestion
	bool lValueOfOrdinaryAssignment = false;
```

the expression is NOT an assigment, it is an identifier on the left side of an assignment.